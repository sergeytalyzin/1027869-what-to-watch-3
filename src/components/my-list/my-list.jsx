import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Operation as DataOperation} from "../../reducer/data/data";
import {getFavoriteList} from "../../reducer/data/selectors.js";
import SmallMovieCard from "../small-movie-card/small-movie-card.jsx";
import withActiveItem from "../../hocs/with-active-item/with-active-item.js";
import {ActionCreator} from "../../reducer/app-status/app-status.js";
import MovieList from "../movie-list/movie-list.jsx";



const MyList = ({favoriteFilmsList, onActiveFilm }) => {

  console.log(favoriteFilmsList);
  return (<div className="user-page">
    <header className="page-header user-page__head">
      <div className="logo">
        <a href="main.html" className="logo__link">
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </a>
      </div>

      <h1 className="page-title user-page__title">My list</h1>

      <div className="user-block">
        <div className="user-block__avatar">
          <img src="img/avatar.jpg" alt="User avatar" width="63" height="63"/>
        </div>
      </div>
    </header>

    <section className="catalog">
      <h2 className="catalog__title visually-hidden">Catalog</h2>

      <div className="catalog__movies-list">


        {

          <MovieList films={favoriteFilmsList} onActiveFilm={onActiveFilm}/>
          // favoriteFilmsList.map((film, i) =>(<article key={i} className="small-movie-card catalog__movies-card">
          //   <div className="small-movie-card__image">
          //     <img src={film.posterBig}
          //       alt={film.title} width="280" height="175"/>
          //   </div>
          //   <h3 className="small-movie-card__title">
          //     <Link className="small-movie-card__link" href="#">{film.title}></Link>
          //   </h3>
          // </article>))
          // favoriteFilmsList.map((it)=>{
          //   return (
          //     <SmallMovieCardWrapper
          //       onActiveFilm = {props.onActiveFilm}
          //       film = {it}
          //       key = {it.id}
          //       active={{}}
          //     />
          //   );
          // })
        }


      </div>
    </section>

    <footer className="page-footer">
      <div className="logo">
        <a href="#" className="logo__link logo__link--light">
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </a>
      </div>

      <div className="copyright">
        <p>Â© 2019 What to watch Ltd.</p>
      </div>
    </footer>
  </div>);
};


MyList.propTypes = {
  onActiveFilm: PropTypes.func.isRequired,
  favoriteFilmsList: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    posterBig: PropTypes.string.isRequired,
  })).isRequired,
};


const mapStateToProps = (state) => ({
  favoriteFilmsList: getFavoriteList(state),
});

const mapStateToDispatch = (dispatch) => ({
  onActiveFilm(film) {
    dispatch(ActionCreator.activeFilm(film));
  }
});


export {MyList};

export default connect(mapStateToProps, mapStateToDispatch)(MyList);
