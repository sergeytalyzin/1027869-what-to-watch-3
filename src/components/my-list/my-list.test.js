import React from "react";
import renderer from "react-test-renderer";
import {MyList} from "./my-list.jsx";


const films = [
  {
    id: 1,
    title: `Mindhunter`,
    src: `somePath`,
    genre: `Comedy`,
    date: 1812,
    posterBig: `img/mindhunter.jpg`,
    rating: 3,
    ratingLevel: `Very good`,
    ratingCount: 111,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.
  Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.`,
    actors: `Frank Sinatra, John Mason, Anthony Mann, Bred Pitt, Heinz Herald`,
    director: `Frank Sinatra`,
    previewVideoLink: `somePath`,
    runTime: `1h 30m`,
  },
  {
    id: 2,
    title: `Mindhunter`,
    src: `somePath`,
    genre: `Comedy`,
    date: 1812,
    posterBig: `img/mindhunter.jpg`,
    rating: 3,
    ratingLevel: `Very good`,
    ratingCount: 111,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.
  Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.`,
    actors: `Frank Sinatra, John Mason, Anthony Mann, Bred Pitt, Heinz Herald`,
    director: `Frank Sinatra`,
    previewVideoLink: `somePath`,
    runTime: `1h 30m`,
  },
];

it(`Should Mylist render correctly`, ()=>{
  const tree = renderer.create(<MyList
    favoriteFilmsList={films}
    onActiveFilm={()=>{}}/>,
  {
    createNodeMock: () => {
      return {};
    }
  }).toJSON();
  expect(tree).toMatchSnapshot();
});
