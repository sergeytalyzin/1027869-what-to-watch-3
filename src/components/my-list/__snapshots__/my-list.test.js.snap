// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Mylist render correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="main.html"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="img/avatar.jpg"
          width="63"
        />
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            alt=""
            className=""
            height={175}
            poster="img/mindhunter.jpg"
            src="somePath"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="#"
          >
            Mindhunter
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            alt=""
            className=""
            height={175}
            poster="img/mindhunter.jpg"
            src="somePath"
            width={280}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="#"
          >
            Mindhunter
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="#"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
